### PR TITLE
Pn/feature edit delete topic comment

### PR DESCRIPTION
**Pull request by nemoipaha**
please see this pull request first, instead of https://github.com/BinaryStudioAcademy/bsa-2016-forum/pull/186 cause maybe it  have conflicts.

### CODE REVIEW

File: resources/assets/js/app/views/comments/TopicCommentNew.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    ui: {
+        'submit': '#submit',
+        'close': '.close'
+    },
+
+    modelEvents: {
+        'invalid': function (model, errors, options) {
+            //logger('model is invalid', errors, options);
+            this.showLoader(false);
+            this.$('.errors').empty();
```
Comment (binaryvo) [inline resources/assets/js/app/views/comments/TopicCommentNew.js]: Is it possible to move $('.errors') to ui property?

File: resources/assets/js/app/views/comments/TopicCommentNew.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            this.showErrors(true);
+        }
+    },
+
+    events: {
+        'click @ui.submit': 'submitComment',
+    },
+
+    showLoader: function (show) {
+        return (show ? this.$('.loader').removeClass('hidden') : this.$('.loader').addClass('hidden'));
```
Comment (binaryvo) [inline resources/assets/js/app/views/comments/TopicCommentNew.js]: Is it possible to move $('.loader') to ui property?

**Comment by binaryvo** [comment]
Please check your code and move elements to ui property

**Comment by nemoipaha** [comment]
Add ui elements.

File: resources/assets/js/app/collections/TopicCommentsCollection.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+var baseCollection = require('../instances/Collection');
+var CommentModel = require('../models/TopicCommentModel');
+
+module.exports = baseCollection.extend({
+    url: '/comments',
+    model: CommentModel,
+
+    collectionEvents: {
```
Comment (binaryvo) [inline resources/assets/js/app/collections/TopicCommentsCollection.js]: There is no collectionEvents property in Collection

**Comment by nemoipaha** [comment]
Fixed!

File: resources/assets/js/app/views/comments/TopicCommentNew.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            }
+        });
+
+        this.showAttachments();
+    },
+
+    filterAttachs: function() {
+        // remove from dropzone some files, that already has uploaded to server
+        this._dropZone.files.filter(function (file) {
+            return file.id == true;
```
Comment (binaryvo) [inline resources/assets/js/app/views/comments/TopicCommentNew.js]: Probably id can't be boolean. You can use condition: file.id > 0

File: app/Http/Controllers/TopicController.php [hunk 5]
```diff
@@ -19,6 +19,35 @@ class TopicController extends ApiController
...
+        $bookmark = $topic->bookmarks()->where('user_id', Auth::user()->id)->first();
+        if ($bookmark !== null) {
+            $data['bookmark'] = $topic->bookmarks()->where('user_id', Auth::user()->id)->first();
+        }
+
+        return $data;
+    }
+
     /**
     /**
```
Comment (binaryvo) [inline app/Http/Controllers/TopicController.php]: Extra comment start

**Pull request closed without merging**